test/dependency_validator/v1_updater_test.dart

```diff
@@ -0,0 +1,317 @@
+// Copyright 2021 Workiva Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'package:over_react_codemod/src/dependency_validator_suggestors/v1_updater.dart';
+import 'package:test/test.dart';
+
+import '../util.dart';
+
+const depValidatorCommand = 'pub run dependency_validator';
+
+Map<String, String Function(String command)> contexts = {
+  'basic-dockerfile': basicDockerFile,
+  'chained-docker-command': chainedDockerfileCommand,
+  'dart-command': dartCommand,
+  'dart-dev-task': dartDevTask,
+  'yaml-script': yamlScript,
+  'yaml-ci-workflow': yamlCiWorkflow,
+  'yaml-skynet': yamlSkynet,
+};
+
+main() {
+  group('V1DependencyUpdater', () {
+    const addedDependency = 'an_added_dependency';
+    final testSuggestor =
+        getSuggestorTester(V1DependencyValidatorUpdater(addedDependency));
+
+    void sharedCommandTests(String context) {
+      group(context, () {
+        final contextWrapper = contexts[context];
+
+        if (contextWrapper == null) {
+          throw ArgumentError('Bad context');
+        }
+
+        test('makes no updates when there is no command', () async {
+          await testSuggestor(
+            shouldDartfmtOutput: false,
+            expectedPatchCount: 0,
+            input: contextWrapper('dart format .'),
```

Comment:
#nit the variable names `context`/`contextWrapper` are a little unclear; what about something like  `contextName`/`getCommandWithinContext` or `testCaseName`/`getSourceFileWithCommand`?